frame/elections/src/lib.rs

```diff
@@ -951,161 +1029,146 @@ impl<T: Config> Pallet<T> {
-							}
-						}
-					}
-					// We then select the new member with the highest weighted stake. In the case of
-					// a tie, the last person in the list with the tied score is selected. This is
-					// the person with the "highest" account id based on the sort above.
-					let prime = prime_votes.into_iter().max_by_key(|x| x.1).map(|x| x.0.clone());
-
-					// new_members_sorted_by_id is sorted by account id.
-					let new_members_ids_sorted = new_members_sorted_by_id
-						.iter()
-						.map(|(m, _)| m.clone())
-						.collect::<Vec<T::AccountId>>();
-
-					// report member changes. We compute diff because we need the outgoing list.
-					let (incoming, outgoing) = T::ChangeMembers::compute_members_diff_sorted(
-						&new_members_ids_sorted,
-						&old_members_ids_sorted,
-					);
-					T::ChangeMembers::change_members_sorted(
-						&incoming,
-						&outgoing,
-						&new_members_ids_sorted,
-					);
-					T::ChangeMembers::set_prime(prime);
-
-					// All candidates/members/runners-up who are no longer retaining a position as a
-					// seat holder will lose their bond.
-					candidates_and_deposit.iter().for_each(|(c, d)| {
-						if new_members_ids_sorted.binary_search(c).is_err() &&
-							new_runners_up_ids_sorted.binary_search(c).is_err()
-						{
-							let (imbalance, _) = T::Currency::slash_reserved(c, *d);
-							T::LoserCandidate::on_unbalanced(imbalance);
-							Self::deposit_event(Event::CandidateSlashed {
-								candidate: c.clone(),
-								amount: *d,
-							});
-						}
-					});
+		Ok(PreElectionResults {
+			num_to_elect,
+			candidate_ids,
+			candidates_and_deposit,
+			voters_and_stakes,
+			voters_and_votes,
+			num_edges,
+		})
+	}
 
-					// write final values to storage.
-					let deposit_of_candidate = |x: &T::AccountId| -> BalanceOf<T> {
-						// defensive-only. This closure is used against the new members and new
-						// runners-up, both of which are phragmen winners and thus must have
-						// deposit.
-						candidates_and_deposit
-							.iter()
-							.find_map(|(c, d)| if c == x { Some(*d) } else { None })
-							.defensive_unwrap_or_default()
-					};
-					// fetch deposits from the one recorded one. This will make sure that a
-					// candidate who submitted candidacy before a change to candidacy deposit will
-					// have the correct amount recorded.
-					<Members<T>>::put(
-						new_members_sorted_by_id
-							.iter()
-							.map(|(who, stake)| SeatHolder {
-								deposit: deposit_of_candidate(who),
-								who: who.clone(),
-								stake: *stake,
-							})
-							.collect::<Vec<_>>(),
-					);
-					<RunnersUp<T>>::put(
-						new_runners_up_sorted_by_rank
-							.into_iter()
-							.map(|(who, stake)| SeatHolder {
-								deposit: deposit_of_candidate(&who),
-								who,
-								stake,
-							})
-							.collect::<Vec<_>>(),
-					);
+	fn do_post_solve_election(
+		winners: Vec<(T::AccountId, u128)>,
+		candidates_and_deposit: Vec<(T::AccountId, BalanceOf<T>)>,
```

Comment:
FYI, these are fairly large pieces of data that we now have to move around, not sure about the exact cost of it, maybe rust does this alloc-free, moving it in and out of stack is probably unavoidable though.